tasks/ModelTask.cpp

```diff
@@ -1,198 +0,0 @@
-gazebo::ModelTask::ModelTask(std::string const& name, TaskCore::TaskState initial_state)
-	: ModelTaskBase(name, initial_state),environment(0)
-{
-}
-//======================================================================================
-gazebo::ModelTask::ModelTask(std::string const& name, RTT::ExecutionEngine* engine, TaskCore::TaskState initial_state)
-	: ModelTaskBase(name, engine, initial_state),environment(0)
-{
-}
-//======================================================================================
-gazebo::ModelTask::~ModelTask()
-{
-	delete joint_port;
-	delete link_port;
-	
-	for(JointPort_V::iterator it = joint_port_list.begin();it != joint_port_list.end(); ++it)
-		delete (it)->first;
-	
-	for(LinkPort_V::iterator it = link_port_list.begin(); it != link_port_list.end(); ++it)
-		delete it->first; 
-	
-	joint_port_list.clear();
-	link_port_list.clear();
-}
-//======================================================================================
-void gazebo::ModelTask::setGazeboModel(physics::WorldPtr _world,  physics::ModelPtr _model, int _environment)
-{
-	world = _world;
-	model = _model;
-	environment = _environment;
-//	sdf = model->GetSDF();
-	
-//	Export Gazebo Joints and Links to Rock-Robotics
-	setJoints();
-	setLinks();
-} 
-//======================================================================================
-void gazebo::ModelTask::setJoints()
-{
-	typedef gazebo::physics::Joint_V Joint_V;
-	Joint_V joints;
-			
-	// Get all joints from a model and creates a Rock component InputPort
-	joints = model->GetJoints();
-	for(Joint_V::iterator joint = joints.begin(); joint != joints.end(); ++joint)
-	{
-		gzmsg <<"RockBridge: found joint: "<< world->GetName() + "/" + model->GetName() + 
-				"/" + (*joint)->GetName() << std::endl;
-		gzmsg << "RockBridge: create joint InputPort in Rock." << std::endl;
-		joint_port = new RTT::InputPort<double>( "joint: " + world->GetName() +
-				"/" + model->GetName() + "/" + (*joint)->GetName() );
-		ports()->addPort( *joint_port );
-		joint_port_list.push_back( std::make_pair(joint_port,*joint) );			
-	}
-	joints.clear();
-}
-//======================================================================================
-void gazebo::ModelTask::setLinks()
-{
-	typedef gazebo::physics::Link_V Link_V;
-	Link_V links; 
-			
-	// Get all links from a model and creates a Rock component InputPort
-	links = model->GetLinks();
-	for(Link_V::iterator link = links.begin(); link != links.end(); ++link)
-	{
-		gzmsg << "RockBridge: found link: "<< world->GetName() + "/" + model->GetName() + 
-				"/" + (*link)->GetName() << std::endl;
-		
-		gzmsg << "RockBridge: create link InputPort in Rock." << std::endl;
-		link_port = new RTT::InputPort<base::Vector3d>( "link: " + world->GetName() +
-				"/" + model->GetName() + "/" + (*link)->GetName() );
-		ports()->addPort(*link_port);
-		link_port_list.push_back( std::make_pair(link_port,*link) );
```

Comment:
Do you mean I should use a gazebo method to read position of links and write them in a Rock output port !?